extensions/ql-vscode/src/model-editor/method-modeling/method-modeling-view-provider.ts

```diff
@@ -100,6 +100,10 @@ export class MethodModelingViewProvider extends AbstractWebviewViewProvider<
           activeState.databaseItem,
           msg.method,
         );
+        this.modelingStore.addModifiedMethod(
```

Comment:
It seems we often have to call `addModifiedMethod(s)` after methods like `updateModeledMethod` and `addModeledMethods`. I wonder if we can move the call to update the modified methods inside the store so that the caller does not have to remember to call both functions?

Reply:
Yes I'd definitely like to move towards that, it's a good idea. In general I can see a few simplifications/improvements we could make around the modeling store. But I think of this as v1, with a lot of features currently in flux, so I'd rather wait a bit before tidying things up. Basically let it evolve a bit and then come back and address some tech debt once we have a fuller picture of its use.